test/no_subscription/shared.dart

```diff
@@ -4,60 +4,59 @@
 
 import 'dart:async';
 
-import 'package:scheduled_test/scheduled_test.dart';
+import 'package:async/async.dart';
+import 'package:test/test.dart';
 import 'package:watcher/watcher.dart';
 
 import '../utils.dart';
 
 void sharedTests() {
-  test('does not notify for changes when there are no subscribers', () {
+  test('does not notify for changes when there are no subscribers', () async {
     // Note that this test doesn't rely as heavily on the test functions in
     // utils.dart because it needs to be very explicit about when the event
     // stream is and is not subscribed.
     var watcher = createWatcher();
+    var queue = new StreamQueue(watcher.events);
 
     // Subscribe to the events.
     var completer = new Completer();
-    var subscription = watcher.events.listen(expectAsync1((event) {
-      expect(event, isWatchEvent(ChangeType.ADD, "file.txt"));
+    queue.next.then((event) {
+      expect(event.type, ChangeType.ADD);
+      expect(event.path.contains("file.txt"), isTrue);
       completer.complete();
-    }));
+    });
```

Comment:
I think you can express this more cleanly as:

```dart
var future = expectLater(queue, emits(isWatchEvent(ChangeType.ADD, "file.txt")));
expect(queue, emitsNothing);
```

The `emitsNothing` assertion does the same thing as `expectAsync1()` in the old code by ensuring no unexpected events are emitted.

Same below.